feature_layers/display-subtype-feature-layer/src/main/java/com/esri/samples/feature_layers/DisplaySubtypeFeatureLayerController.java

```diff
@@ -0,0 +1,143 @@
+  @FXML private CheckBox checkBox;
+  private SubtypeSublayer sublayer;
+  private Renderer originalRenderer;
+  
+  public void initialize() {
+
+    try {
+
+      // create a map with streets night vector basemap and add it to the map view
+      ArcGISMap map = new ArcGISMap();
+      map.setBasemap(Basemap.createStreetsNightVector());
+      mapView.setMap(map);
+      // display the current map scale 
+      mapView.addMapScaleChangedListener(mapScaleChangedEvent ->
+        currentMapScaleLabel.setText("Current Map Scale: 1:" + Math.round(mapView.getMapScale())));
+
+      // set the viewpoint to Naperville, Illinois
+      Viewpoint initialViewpoint = new Viewpoint(new Envelope(-9812691.11079696, 5128687.20710657,
+        -9812377.9447607, 5128865.36767282, SpatialReferences.getWebMercator()));
+      map.setInitialViewpoint(initialViewpoint);
+
+      // create a subtype feature layer from the service feature table, and add it to the map
+      final String serviceFeatureTableUrl = "https://sampleserver7.arcgisonline" +
+        ".com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/100";
+      ServiceFeatureTable serviceFeatureTable = new ServiceFeatureTable(serviceFeatureTableUrl);
+      SubtypeFeatureLayer subtypeFeatureLayer = new SubtypeFeatureLayer(serviceFeatureTable);
+      map.getOperationalLayers().add(subtypeFeatureLayer);
+      
+      // access the json required for sublayer label definitions
+      File jsonFile = new File(System.getProperty("data.dir"),
+          Objects.requireNonNull(getClass().getClassLoader().getResource("label_definition.json")).getFile());
+      final String json;
+      // read in the complete file as a string
+      try (Scanner scanner = new Scanner(jsonFile)) {
+        json = scanner.useDelimiter("\\A").next();
+      }
+
+      subtypeFeatureLayer.loadAsync();
+      subtypeFeatureLayer.addDoneLoadingListener(() -> {
+        
+        // get the Street Light sublayer and define its labels
+        sublayer = subtypeFeatureLayer.getSublayerWithSubtypeName("Street Light");
+        sublayer.setLabelsEnabled(true);
+        sublayer.getLabelDefinitions().add(LabelDefinition.fromJson(json));
+
+        // get the original renderer of the sublayer
+        originalRenderer = sublayer.getRenderer();
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Sets the visibility of the sublayer.
+   */
+  @FXML
+  private void controlSublayerVisibility() {
+    sublayer.setVisible(checkBox.isSelected());
+  }
+  
+  /**
+   * Switches between a renderer with a pink diamond symbol and the sublayer's original renderer.
+   */
+  @FXML
+  private void handleToggleRendererButtonClicked() {
+    if (toggleRendererButton.isSelected()) {
+      Symbol symbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.DIAMOND, 0xfff58f84, 20);
+      Renderer sublayerRenderer = new SimpleRenderer(symbol);
```

Comment:
To be a bit more resource efficient and avoid creating a new `Renderer` every time this is called, you could create the Symbol and Renderer in the `initialize()` and then simply apply it in this method.